internal/cmd/operator-sdk/bundle/validate/optional.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	apimanifests "github.com/operator-framework/api/pkg/manifests"
+	apivalidation "github.com/operator-framework/api/pkg/validation"
+	apierrors "github.com/operator-framework/api/pkg/validation/errors"
+	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// Keys for label selectors to be used by all validators.
+const (
+	nameKey  = "name"
+	suiteKey = "suite"
+)
+
+// optionalValidators is a list of validators with 0their name, labels for CLI usage, and a light description.
+var optionalValidators = validators{
+	{
+		Validator: apivalidation.OperatorHubValidator,
+		name:      "operatorhub",
+		labels: map[string]string{
+			nameKey:  "operatorhub",
+			suiteKey: "operatorframework",
+		},
+		desc: "OperatorHub.io metadata validation",
+	},
+}
+
+// runOptionalValidators runs optional validators selected by sel on bundle.
+func runOptionalValidators(bundle *apimanifests.Bundle, sel labels.Selector) []apierrors.ManifestResult {
```

Comment:
WDYT about we try to persuade here the DDD principle?
See that we can have a BundleValidationOptional structure and then:

(b *BundleValidationOptional) run()
(b *BundleValidationOptional) list()

Reply:
These functions are wrappers for the exact design you suggest (methods on `validators`).